### PR TITLE
GitHub Actions

### PR DESCRIPTION
**Pull request by amineoualialami**

### CODE REVIEW

File: posts/deployToEcsWithGithubAction.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+- Amazon ECS
+- AWS Fargate
+- Amazon ECR
+
+date: 2020-20-10T20:16:10.000Z
+title: "Deploy to ECS using GitHub Actions"
+image: https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2020/01/deployToEcsWithGithubAction.jpg
+---
+
+Building a DevOps platform nowadays can be challenging for Architect and Infrastructure teams, due to the integration complexity with many platforms: source control, ci/cd, artifact repositories, quality, SAST/SCA, deployment ...  , Many DevOps leader believes that the challenge of DevOps is to reduce the number of the platforms involved in one pipeline.
```
Comment (raphaelbrugier) [inline posts/deployToEcsWithGithubAction.md]:
```suggestion
Building a DevOps platform nowadays can be challenging for Architect and Infrastructure teams, due to the integration complexity with many platforms: source control, ci/cd, artifact repositories, quality, SAST/SCA, deployment ...  , Many DevOps leader believes that the challenge of DevOps is to reduce the number of platforms involved in one pipeline.
```

File: posts/deployToEcsWithGithubAction.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+
+# Create your first pipeline with Github actions
+On your GitHub repository select the Actions tab.
+GitHub propose popular pipelines to start with, we will choose: Deploy to Amazon ECS
+
+![01](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2020/01/deployToEcsWithGithubAction-01.png)
+
+It will create in your source code the file /.github/workflows/aws.yml that represents your GitHub Actions.
+We added two more steps to set up the JDK 8 and the maven command to build the Spring Boot JAR file.
```
Comment (raphaelbrugier) [inline posts/deployToEcsWithGithubAction.md]:
```suggestion
We add two more steps to set up the JDK 8 and the maven command to build the Spring Boot JAR file.
```

Comment (raphaelbrugier) [inline posts/deployToEcsWithGithubAction.md]: JDK 8 is old now! It's for dinosaurs like us! :trollface:  If it's not too much work and impact, maybe you can update the example to use the latest?

File: posts/deployToEcsWithGithubAction.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+```
+
+After running this pipeline, the results looks like :
+
+![02](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2020/01/deployToEcsWithGithubAction-02.png)
+
+
+
+# Conclusion
+GitHub is working actively to integrate all the necessary DevOps features in their platform like GitHub Package to store your artifacts, and security alerts that track security vulnerabilities that apply to your repository dependencies (SCA).
```
Comment (raphaelbrugier) [inline posts/deployToEcsWithGithubAction.md]: Maybe add a link to SCA? I personally didn't know this acronym.

Comment (yhippa) [reply posts/deployToEcsWithGithubAction.md]: Definitely want to explain that: https://www.quickanddirtytips.com/education/grammar/how-to-define-abbreviations-in-a-document

File: posts/deployToEcsWithGithubAction.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+```js
+FROM openjdk:8-jdk-alpine
+RUN apk --no-cache add curl
+VOLUME /tmp
+EXPOSE 8888
+ADD target/dcdojo-0.0.1-SNAPSHOT.jar app.jar
+ENTRYPOINT ["java","-Djava.security.egd=file:/dev/./urandom","-jar","app.jar"]
+```
+
+After running this pipeline, the results looks like :
```
Comment (yhippa) [inline posts/deployToEcsWithGithubAction.md]:
```suggestion
After running this pipeline, the results looks like:
```

File: posts/deployToEcsWithGithubAction.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+- Amazon ECS
+- AWS Fargate
+- Amazon ECR
+
+date: 2020-20-10T20:16:10.000Z
+title: "Deploy to ECS using GitHub Actions"
+image: https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2020/01/deployToEcsWithGithubAction.jpg
+---
+
+Building a DevOps platform nowadays can be challenging for architecture and plateform infrastructure teams, due to the integration complexity with many platforms: source control, ci/cd, artifact repositories, quality, SAST/SCA, deployment ...  , Many DevOps leader believes that the challenge of DevOps is to reduce the number of platforms involved in one pipeline.
```
Comment (nnightingale14) [inline posts/deployToEcsWithGithubAction.md]: Building a DevOps platform nowadays can become a challenge for architecture and platform infrastructure teams, due to the integration complexity across several instances: Source Control, CI/CD, Artifact Repositories, Quality, SAST/SCA, Deployment, etc.  Many DevOps leaders believe that the challenge of DevOps is to ultimately reduce the number of platforms involved in one single pipeline.

File: posts/deployToEcsWithGithubAction.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+That's why tech giants are trying to integrate all the pipeline steps into their platforms.
```
Comment (nnightingale14) [inline posts/deployToEcsWithGithubAction.md]: That's why tech giants are attempting to integrate **_all_** pipeline steps into their platforms.

File: posts/deployToEcsWithGithubAction.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+In this article, we will deploy a simple [Spring Boot](https://spring.io/projects/spring-boot) application to [Amazon Elastic Container Service (ECS)](https://spring.io/projects/spring-boot) using the new GitHub feature called [GitHub Actions](https://github.com/features/actions).
+
+# What is GitHub Actions
```
Comment (nnightingale14) [inline posts/deployToEcsWithGithubAction.md]: # What is GitHub Actions?

File: posts/deployToEcsWithGithubAction.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+GitHub Actions enables you to create custom software development lifecycle pipelines directly in your Github repository.
```
Comment (nnightingale14) [inline posts/deployToEcsWithGithubAction.md]: *GitHub

File: posts/deployToEcsWithGithubAction.md [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+You can write individual tasks, called actions, and combine them to create a custom pipeline.
```
Comment (nnightingale14) [inline posts/deployToEcsWithGithubAction.md]: * called "actions,"

File: posts/deployToEcsWithGithubAction.md [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+Behind the scenes, your pipeline is run on GitHub-hosted machines, called runners.
```
Comment (nnightingale14) [inline posts/deployToEcsWithGithubAction.md]: * called "runners."

File: posts/deployToEcsWithGithubAction.md [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+
+
+# Create your first pipeline with Github actions
```
Comment (nnightingale14) [inline posts/deployToEcsWithGithubAction.md]: Create Your First Pipeline With GitHub Actions

File: posts/deployToEcsWithGithubAction.md [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
+On your GitHub repository select the Actions tab.
```
Comment (nnightingale14) [inline posts/deployToEcsWithGithubAction.md]: * "Actions"

File: posts/deployToEcsWithGithubAction.md [hunk 13]
```diff
@@ -0,0 +1,177 @@
...
+GitHub propose popular pipelines to start with. We will choose: Deploy to Amazon ECS
```
Comment (nnightingale14) [inline posts/deployToEcsWithGithubAction.md]: GitHub proposes popular pipelines right out of the gate. We will choose: **Deploy to Amazon ECS**

File: posts/deployToEcsWithGithubAction.md [hunk 14]
```diff
@@ -0,0 +1,177 @@
...
+        wait-for-service-stability: true
+
+```
+
+This pipeline  will checkout the code, build the JAR file, build and push the Docker image to [Amazon Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/), and then  deploy a new task definition to Amazon ECS, on every push to the master branch.
+
+Before using this pipeline, you will need to complete all the necessary infrastructure configuration on AWS:
+- Create an ECS repository
+- Create an ECS task definition, an ECS cluster, and an ECS service. [ECS getting started](https://us-east-2.console.aws.amazon.com/ecs/home?region=us-east-2#/firstRun)
+- Store your ECS task definition as a Json file in your repository
```
Comment (nnightingale14) [inline posts/deployToEcsWithGithubAction.md]: JSON

File: posts/deployToEcsWithGithubAction.md [hunk 15]
```diff
@@ -0,0 +1,177 @@
...
+```js
+FROM openjdk:8-jdk-alpine
+RUN apk --no-cache add curl
+VOLUME /tmp
+EXPOSE 8888
+ADD target/dcdojo-0.0.1-SNAPSHOT.jar app.jar
+ENTRYPOINT ["java","-Djava.security.egd=file:/dev/./urandom","-jar","app.jar"]
+```
+
+After running this pipeline, the results looks like:
```
Comment (nnightingale14) [inline posts/deployToEcsWithGithubAction.md]: * look like

File: posts/deployToEcsWithGithubAction.md [hunk 16]
```diff
@@ -0,0 +1,177 @@
...
+
+![02](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2020/01/deployToEcsWithGithubAction-02.png)
+
+
+
+# Conclusion
+GitHub is working actively to integrate all the necessary DevOps features in their platform like GitHub Package to store your artifacts, and security alerts that track security vulnerabilities that apply to your project dependencies (Sofware Composition Analysis).
```
Comment (nnightingale14) [inline posts/deployToEcsWithGithubAction.md]: GitHub is actively working to integrate all necessary DevOps features in their platform, such as GitHub Package, which stores your artifacts and security alerts that track security vulnerabilities that apply to your project dependencies (Software Composition Analysis).

**Pull request merged**